ink-wrapper-types/src/drink/mod.rs

```diff
@@ -0,0 +1,69 @@
+mod client;
+
+use ::drink::{frame_system, runtime::HashFor, DispatchError, Weight};
+pub use client::*;
+
+use crate::{ContractEvent, ExecCall, InstantiateCall, ReadCall, UploadCall};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Decoding error: {0}")]
+    DecodingError(String),
+    #[error("Upload failed")]
+    UploadFailed,
+    #[error("Code hash mismatch")]
+    CodeHashMismatch,
+    #[error("Deployment reverted")]
+    DeploymentReverted,
+    #[error("Deployment failed: {0:?}")]
+    DeploymentFailed(DispatchError),
+    #[error("Contract call failed: {0:?}")]
+    CallFailed(DispatchError),
+}
+
+pub trait Connection<R: frame_system::Config> {
```

Comment:
this trait should be rather universal for the whole `ink-wrapper-types` crate, right? not just the `drink` backend

Reply:
For ``aleph-client` it was async - since every method required communicating with the node. It'll be the same for `subxt`.